packages/components/codemods/migrateGuidanceBlockActionsToActionsSlot/transformActionsToActionsSlot.spec.ts

```diff
@@ -0,0 +1,196 @@
+    const inputAst = parseJsx(`
+      <GuidanceBlock
+        layout="default"
+        illustration={<Informative alt="" />}
+        content={<div>Test</div>}
+        actions={{
+          primary: {
+            label: 'Primary action',
+            onClick: () => alert('click 1'),
+          },
+          secondary: {
+            label: 'Secondary action',
+            href: "#secondary"
+          },
+        }}
+      />`)
+    const outputAst = parseJsx(`
+      <GuidanceBlock
+        layout="default"
+        illustration={<Informative alt="" />}
+        content={<div>Test</div>}
+        actionsSlot={<><Button onPress={() => alert('click 1')} variant="secondary" size="large">Primary action</Button><LinkButton href="#secondary" variant="tertiary" size="large">Secondary action</LinkButton></>}
+        />
+      `)
+
+    expect(transformInput(inputAst)).toBe(printAst(outputAst))
+  })
+  it('transforms a primary action with all v1 Button props to expected Button outputs', () => {
+    const inputAst = parseJsx(`
+      <GuidanceBlock
+        layout="default"
+        illustration={<Informative alt="" />}
+        content={<div>Test</div>}
+        actions={{
+          primary: {
+            label: 'Learn more',
+            onClick: () => alert('tada: 🎉'),
+            tooltip: {
+              text: 'Opens in a new tab',
+              mood: 'cautionary',
+            },
+            badge: {
+              text: 'New',
+            },
+            destructive: true,
+            disabled: hasCondition ? true : false,
+            reversed: true,
+            icon: <Icon name="arrow_forward" shouldMirrorInRTL isPresentational />,
+            iconPosition: 'end',
+            size: 'small',
+            working: true,
+            workingLabel: 'Loading...',
+            workingLabelHidden: true,
+            disableTabFocusAndIUnderstandTheAccessibilityImplications: true,
+            'data-custom-attr': 'custom-attr',
+          },
+        }}
+      />`)
+    const outputAst = parseJsx(`
+        <GuidanceBlock
+          layout="default"
+          illustration={<Informative alt=""/>}
+          content={<div>Test</div>}
+          actionsSlot={<><Button
+              onPress={() => alert('tada: 🎉')}
+              tooltip={{text: 'Opens in a new tab',mood: 'cautionary',}}
+              badge={{ text: 'New', }}
+              isDisabled={hasCondition ? true : false}
+              isReversed
+              icon={<Icon name="arrow_forward" shouldMirrorInRTL isPresentational/>}
+              iconPosition='end'
+              size="medium"
+              isPending pendingLabel='Loading...'
+              hasHiddenPendingLabel
+              data-custom-attr='custom-attr'
+              variant="secondary"
+            >Learn more</Button></>}/>
+      `)
+
+    expect(transformInput(inputAst).replace(/\n+/g, ' ').replace(/\s+/g, ' ')).toBe(
+      printAst(outputAst).replace(/\n+/g, ' ').replace(/\s+/g, ' '),
```

Comment:
`\s+` is equivalent to looking for any of these `\r\n\t\f\v` so you only need to do the one global replace

```suggestion
    expect(transformInput(inputAst).replace(/\s+/g, ' ')).toBe(printAst(outputAst).replace(/\s+/g, ' '),
```

Reply:
Ah that sweet regex magic I always forget 🪄 🐇